Runtime/Scripts/RGBotConfigs/RGEntity.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace RegressionGames.RGBotConfigs
+{
+    public class RGEntity : MonoBehaviour
+    {
+        [Header("General Information")] [Tooltip("Does this object represent a human/bot player ?")]
+        public bool isPlayer;
+
+        [Tooltip("A type name for associating like objects in the state")]
+        public string objectType;
+
+        // this is used in our toolkit to understand which things would need dynamic models
+        [Tooltip("Is this object spawned during runtime, or a fixed object in the scene?")]
+        public bool isRuntimeObject = false;
+
+        [Header("3D Positioning")] 
+        public bool syncPosition = true;
+        public bool syncRotation = true;
+        
+        // maps action names to RGAction components
+        private Dictionary<string, RGAction> actionMap = new Dictionary<string, RGAction>();
+            
+        // The client Id that owns this agent
+        // Used as a performance optimization for mapping the ClientId into the state payloads
+        public long? ClientId = null;
+        
+        /**
+     * Updates the registry with a game object that has RGAction scripts attached to it
+     */
+        void Start()
+        {
+            var actions = GetComponents<RGAction>();
+            foreach (var action in actions)
+            {
+                actionMap[action.GetActionName()] = action;
+            }
+
+            RGDebug.LogDebug($"Agent registered with {actionMap.Count} actions");
```

Comment:
So in my mental model (correct me if I am wrong), we have entities, some of which may be agents (i.e. the ones with actions are agents I suppose). If that is correct, maybe we only print this if there is at least one action? That way entities with no actions don't have a console log that say the word "Agent"

Reply:
That's a solid way to think about it. Still I think because we're depreciating references to RGAgent, using 'agent' as meaning 'entity with at least one action' might be confusing once 'agent' is no longer an actual class or separate object. In cases like this perhaps a better approach is 

```
if (actionMap.Count > 0)
{
    RGDebug.LogDebug($"Entity registered with {actionMap.Count} actions");
}
```

Reply:
That makes a lot of sense to me! Happy to not add extra points of confusion :+1: